thumbnail-service/src/main/java/no/sikt/nva/handler/ThumbnailRequestHandler.java

```diff
@@ -0,0 +1,144 @@
+    public static final String FILENAME_PREFIX = "filename=\"";
+    public static final String FILENAME_POSTFIX = "\"";
+    private static final Logger logger = LoggerFactory.getLogger(ThumbnailRequestHandler.class);
+
+    // protects against overwriting existing files in
+    // directory, plus it's the only place aws allows filewriting; To view thumbnails created locally: replace with
+    // "" and comment out outputFile.deleteOnExit() / inputFile.deleteOnExit();
+    private static final String INPUT_FILE_NAME_PREFIX = "/tmp/";
+    private static final String OUTPUT_FILE_NAME = "thumbnail.png";
+    private static final String THUMBNAIL_BUCKET_ENVIRONMENT_FIELD = "THUMBNAIL_BUCKET";
+    public static final String DELIMITER = ", ";
+    private final String thumbnailBucketName;
+    private final S3Client s3Client;
+    private File inputFile;
+    private String mimeTypeFromS3Response;
+
+    @JacocoGenerated
+    public ThumbnailRequestHandler() {
+        this(S3Driver.defaultS3Client().build());
+    }
+
+    public ThumbnailRequestHandler(S3Client s3Client) {
+        this.thumbnailBucketName = new Environment().readEnv(THUMBNAIL_BUCKET_ENVIRONMENT_FIELD);
+        this.s3Client = s3Client;
+    }
+
+    @Override
+    public Void handleRequest(S3Event s3Event, Context context) {
+        inputFile = readFile(s3Event);
+        var outputFile = generateThumbnail();
+        writeThumbnailToS3(outputFile);
+        outputFile.deleteOnExit();
+        inputFile.deleteOnExit();
+        return null;
+    }
+
+    private void writeThumbnailToS3(File outputFile) {
+        try {
+            s3Client.putObject(PutObjectRequest.builder()
+                                   .bucket(thumbnailBucketName)
+                                   .key("keyForFeature(featureState.getFeature())")
+                                   .build(), RequestBody.fromFile(outputFile));
+        } catch (RuntimeException e) {
+            logger.warn(e.getMessage());
+            throw new RuntimeException(e);
+        }
+    }
+
+    private File generateThumbnail() {
+        try (var thumbnailerManager = new ThumbnailerManager()) {
+            logCurrentSupportedMimeTypes(thumbnailerManager);
+            var outPutFile = new File(INPUT_FILE_NAME_PREFIX + OUTPUT_FILE_NAME);
+            thumbnailerManager.generateThumbnail(inputFile, outPutFile, mimeTypeFromS3Response);
+            return outPutFile;
+        } catch (IOException | ThumbnailerException e) {
+            logger.warn(COULD_NOT_CREATE_THUMBNAIL_LOG_MESSAGE);
+            throw new RuntimeException(e);
+        }
+    }
+
+    private void logCurrentSupportedMimeTypes(ThumbnailerManager thumbnailerManager) {
+        var supportedMimeTypes = String.join(DELIMITER, thumbnailerManager.getAcceptedMimeTypes());
+        logger.info(CURRENTLY_SUPPORTING_THE_FOLLOWING_MIME_TYPES_LOG_MESSAGE + supportedMimeTypes);
```

Comment:
Why do we need to log all currently supported Mime types for every call?

Reply:
According to documentation of the old thumbnailer: what mime-types that is supported depends on the JAVA runtime installed. This is a weakness in my opinion, as this would create a situation where it suddenly stops supporting formats from one day to another.Plus the classic "it runs on my PC"

Reply:
But I can change this to only log at debug level or only log if we get a ThumbnailerException.

Reply:
change to only debug level